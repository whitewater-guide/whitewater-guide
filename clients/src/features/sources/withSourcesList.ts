import { graphql } from 'react-apollo';
import { Connection, Source } from '../../../ww-commons';
import { queryResultToList, WithList } from '../../apollo';
import listSources from './listSources.query';

interface Result {
  sources: Connection<Source>;
}

export interface WithSourcesList {
  sources: WithList<Source>;
}

export const withSourcesList = graphql<Result, any, WithSourcesList>(
  listSources,
  {
    alias: 'withSourcesList',
    options: {
      fetchPolicy: 'cache-and-network',
    },
    props: props => queryResultToList(props, 'sources'),
  },
);
