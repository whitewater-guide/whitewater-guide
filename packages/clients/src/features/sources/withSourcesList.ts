import { Connection, Source } from '@whitewater-guide/commons';
import { graphql } from 'react-apollo';
import { queryResultToList, WithList } from '../../apollo';
import listSources from './listSources.query';

interface Result {
  sources: Connection<Source>;
}

export interface WithSourcesList {
  sources: WithList<Source>;
}

export const withSourcesList = graphql<any, Result, any, WithSourcesList>(
  listSources,
  {
    alias: 'withSourcesList',
    options: () => ({
      fetchPolicy: 'cache-and-network',
    }),
    props: (props) => queryResultToList(props, 'sources'),
  },
);
