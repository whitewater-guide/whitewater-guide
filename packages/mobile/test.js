const MarkdownIt = require('markdown-it');
const md = new MarkdownIt();
const res = MarkdownIt.parseInline(
  'The content you are adding will be available under an open licence (CC). See our [Terms of Use](1) and others',
);

const nodes = [];
let currentLink;
res[0].children.forEach((child) => {
  if (child.type === 'text') {
    if (currentLink) {
      currentLink.text = child.content;
    } else {
      nodes.push(child.content);
    }
  } else if (child.type === 'link_open') {
    currentLink = { href: child.attrGet('href') };
  } else if (child.type === 'link_close') {
    nodes.push(currentLink);
    currentLink = null;
  }
});
console.log(nodes);
