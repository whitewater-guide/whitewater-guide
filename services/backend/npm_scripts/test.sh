#!/usr/bin/env bash

set -o allexport
source ../../config/.env.test
source .env.test
set +o allexport

./node_modules/.bin/jest -i --forceExit "$@"
