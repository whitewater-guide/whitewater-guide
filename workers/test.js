const { execFile } = require('child_process');

execFile('./bin/norway', ['harvest', '--code=213.4', '--html'], (error, stdout, stderr) => {
  if (error) {
    throw error;
  }
  const result = JSON.parse(stdout)
  console.log(JSON.stringify(result, null, 2))
  // console.log(result.success, result.data.length)
});



