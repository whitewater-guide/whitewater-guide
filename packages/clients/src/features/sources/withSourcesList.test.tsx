import { mount } from 'enzyme';
import React from 'react';
import { createMockedProvider, flushPromises, Receiver } from '../../test';
import { WithSourcesList, withSourcesList } from './withSourcesList';

let receiver: Receiver<WithSourcesList>;

beforeEach(async () => {
  const MockedProvider = createMockedProvider();
  const WithData = withSourcesList(Receiver as any);
  const wrapped = mount(
    <MockedProvider>
      <WithData />
    </MockedProvider>,
  );
  receiver = wrapped
    .find(Receiver as any)
    .first()
    .instance() as any;
  await flushPromises();
});

it('should have loading state', async () => {
  expect(receiver.cwrp[0]).toMatchObject({
    sources: {
      nodes: [],
      loading: true,
      count: 0,
    },
  });
  expect(receiver.cwrp[1].sources.loading).toBe(false);
});

it('should match snapshot', async () => {
  expect(receiver.cwrp[0].sources.nodes).toHaveLength(0);
  expect(receiver.cwrp[1].sources).toMatchSnapshot();
});
