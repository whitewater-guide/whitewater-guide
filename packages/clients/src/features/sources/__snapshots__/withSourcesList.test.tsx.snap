// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "count": -413,
  "error": undefined,
  "fetchMore": [Function],
  "loading": false,
  "networkStatus": 7,
  "nodes": Array [
    Object {
      "__typename": "Source",
      "cron": "Porro delectus et",
      "enabled": false,
      "gauges": Object {
        "__typename": "SourceGaugeConnection",
        "count": -373,
        Symbol(id): "$Source:7fbe024f-3316-4265-a6e8-c65a837e308a:Esse blanditiis natus.gauges",
      },
      "harvestMode": "oneByOne",
      "id": "7fbe024f-3316-4265-a6e8-c65a837e308a",
      "language": "Esse blanditiis natus",
      "name": "Qui itaque",
      "status": Object {
        "__typename": "HarvestStatus",
        "count": 385,
        "success": true,
        "timestamp": 2017-01-31T21:00:00.000Z,
        Symbol(id): "$Source:7fbe024f-3316-4265-a6e8-c65a837e308a:Esse blanditiis natus.status",
      },
      Symbol(id): "Source:7fbe024f-3316-4265-a6e8-c65a837e308a:Esse blanditiis natus",
    },
    Object {
      "__typename": "Source",
      "cron": "Architecto ducimus",
      "enabled": true,
      "gauges": Object {
        "__typename": "SourceGaugeConnection",
        "count": -62,
        Symbol(id): "$Source:9eded1d0-520e-4a19-9af6-93b452acd600:Fuga maiores distinctio.gauges",
      },
      "harvestMode": "allAtOnce",
      "id": "9eded1d0-520e-4a19-9af6-93b452acd600",
      "language": "Fuga maiores distinctio",
      "name": "Ullam veniam ducimus",
      "status": Object {
        "__typename": "HarvestStatus",
        "count": -808,
        "success": false,
        "timestamp": 2017-01-31T21:00:00.000Z,
        Symbol(id): "$Source:9eded1d0-520e-4a19-9af6-93b452acd600:Fuga maiores distinctio.status",
      },
      Symbol(id): "Source:9eded1d0-520e-4a19-9af6-93b452acd600:Fuga maiores distinctio",
    },
  ],
  "refetch": [Function],
}
`;
